[Issue] - Container App without ingress (background worker) fails to deploy

**Output from `azd version`**
azd version 0.4.0-beta.1 (commit 2bf7a529914d86d8e9f5d7b6f95a4f20a2351e01)

**Output from `az version`** 
{
  "azure-cli": "2.42.0",
  "azure-cli-core": "2.42.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {
    "containerapp": "0.3.14",
    "logic": "0.1.6"
  }
}

**Describe the bug**
azd deploy fails with container app if/when container app does not have an ingress defined; for example, a background worker may not need or want an ingress

Error:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x50 pc=0x63e0a8]

goroutine 15 [running]:
github.com/azure/azure-dev/cli/azd/pkg/tools/azcli.(*azCli).GetContainerAppProperties(0x831ca0?, {0xa1fa30, 0xc000405f80}, {0xc000099080?, 0xc0000af678?}, {0xc000096aba, 0x6}, {0xc00011e8e8, 0x13})
        D:/a/_work/1/s/cli/azd/pkg/tools/azcli/container_app.go:29 +0x1a8
github.com/azure/azure-dev/cli/azd/pkg/project.(*containerAppTarget).Endpoints(0xc000160c80, {0xa1fa30, 0xc000405f80})
        D:/a/_work/1/s/cli/azd/pkg/project/service_target_containerapp.go:153 +0xbf
github.com/azure/azure-dev/cli/azd/pkg/project.(*containerAppTarget).Deploy(0xc000160c80, {0xa1fa30, 0xc000405f80}, 0x25201d?, {0xc0002e34a0, 0x47}, 0xc0001066c0?)
        D:/a/_work/1/s/cli/azd/pkg/project/service_target_containerapp.go:135 +0x13ef
github.com/azure/azure-dev/cli/azd/pkg/project.(*Service).Deploy.func1()
        D:/a/_work/1/s/cli/azd/pkg/project/service.go:71 +0x3c2
created by github.com/azure/azure-dev/cli/azd/pkg/project.(*Service).Deploy
        D:/a/_work/1/s/cli/azd/pkg/project/service.go:52 +0x10c

**To Reproduce**
Create a container app that does not have an ingress defined

**Expected behavior**
azd deploy should work without error

**Environment**
Information on your environment:
    * dotnet 7/csharp
    * vscode

**Additional context**
Snippet from problem bicep file:

resource messageRouterApp 'Microsoft.App/containerApps@2022-06-01-preview' = {
  name: '${name}messagerouterapp'
  location: location
  tags: tags
  identity: {
    type: 'UserAssigned'
    userAssignedIdentities: {
      '${managedIdentity.id}': {}
    }
  }
  properties: {
    managedEnvironmentId: containerAppEnvironment.id
    configuration: {
      registries: [
        {
          identity: managedIdentity.id
          server: containerRegistry.properties.loginServer
        }
      ]
    }
    template: {
    ...

Adding the following to the app's configuration fixes the problem.
ingress: {
   external: true
   targetPort: 5075
   transport: 'auto'
}

The problem code appears here https://github.com/Azure/azure-dev/blob/main/cli/azd/pkg/tools/azcli/container_app.go on lines 28-29.  It should protect against this value being null.

--- a/cli/azd/pkg/tools/azcli/container_app.go
+++ b/cli/azd/pkg/tools/azcli/container_app.go
@@ -24,9 +24,19 @@ func (cli *azCli) GetContainerAppProperties(
 	if err != nil {
 		return nil, fmt.Errorf("failed retrieving container app properties: %w", err)
 	}
+	
+	var hostNames []string
+	if containerApp.Properties != nil
+		&& containerApp.Properties.Configuration != nil
+		&& containerApp.Properties.Configuration.Ingress != nil
+		&& containerApp.Properties.Configuration.Ingress.Fqdn != nil {
+		hostNames = []string{*containerApp.Properties.Configuration.Ingress.Fqdn }
+	} else {
+		hostNames = []string{}
+	}
 
 	return &AzCliContainerAppProperties{
-		HostNames: []string{*containerApp.Properties.Configuration.Ingress.Fqdn},
+		HostNames: hostNames,
 	}, nil
 }
 

--- a/cli/azd/pkg/tools/azcli/container_app.go
+++ b/cli/azd/pkg/tools/azcli/container_app.go
@@ -24,13 +24,13 @@ func (cli *azCli) GetContainerAppProperties(
 	if err != nil {
 		return nil, fmt.Errorf("failed retrieving container app properties: %w", err)
 	}
-	
+
 	var hostNames []string
-	if containerApp.Properties != nil
-		&& containerApp.Properties.Configuration != nil
-		&& containerApp.Properties.Configuration.Ingress != nil
-		&& containerApp.Properties.Configuration.Ingress.Fqdn != nil {
-		hostNames = []string{*containerApp.Properties.Configuration.Ingress.Fqdn }
+	if containerApp.Properties != nil &&
+		containerApp.Properties.Configuration != nil &&
+		containerApp.Properties.Configuration.Ingress != nil &&
+		containerApp.Properties.Configuration.Ingress.Fqdn != nil {
+		hostNames = []string{*containerApp.Properties.Configuration.Ingress.Fqdn}
 	} else {
 		hostNames = []string{}
 	}

